File: testers/avro/java-json/generated-and-checked-in/com/example/events/OrderEvents.java
package com.example.events;

import com.fasterxml.jackson.annotation.JsonSubTypes;
import com.fasterxml.jackson.annotation.JsonSubTypes.Type;
import com.fasterxml.jackson.annotation.JsonTypeInfo;

@JsonTypeInfo(use = JsonTypeInfo.Id.NAME, include = JsonTypeInfo.As.PROPERTY, property = "@type")
@JsonSubTypes(
    value = {
      @Type(value = OrderCancelled.class, name = "OrderCancelled"),
      @Type(value = OrderPlaced.class, name = "OrderPlaced"),
      @Type(value = OrderUpdated.class, name = "OrderUpdated")
    })
public sealed interface OrderEvents permits OrderCancelled, OrderPlaced, OrderUpdated {}

File: testers/avro/scala-json/generated-and-checked-in/com/example/events/OrderEvents.scala
package com.example.events

import com.fasterxml.jackson.annotation.JsonSubTypes
import com.fasterxml.jackson.annotation.JsonSubTypes.Type
import com.fasterxml.jackson.annotation.JsonTypeInfo

@JsonTypeInfo(use = JsonTypeInfo.Id.NAME, include = JsonTypeInfo.As.PROPERTY, property = "@type")
@JsonSubTypes(value = Array(new Type(value = classOf[OrderCancelled], name = "OrderCancelled"), new Type(value = classOf[OrderPlaced], name = "OrderPlaced"), new Type(value = classOf[OrderUpdated], name = "OrderUpdated")))
trait OrderEvents